plugins/pulp_rpm/plugins/importers/yum/sync.py

```diff
@@ -207,16 +269,16 @@ def _progress_summary(self):
             ret[step_name] = {'state': progress_dict['state']}
         return ret
 
-    def get_metadata(self):
+    def check_metadata(self, url):
         """
-        :return:    instance of MetadataFiles where each relevant file has been
-                    identified and downloaded.
+        :param url: curret URL we should sync
+        :type: str
+
+        :return:    instance of MetadataFiles
         :rtype:     pulp_rpm.plugins.importers.yum.repomd.metadata.MetadataFiles
         """
-        _logger.info(_('Downloading metadata from %(feed)s.') % {'feed': self.sync_feed})
-        metadata_files = metadata.MetadataFiles(self.sync_feed, self.tmp_dir, self.nectar_config)
-        # allow the downloader to be accessed by the cancel method if necessary
-        self.downloader = metadata_files.downloader
```

Comment:
I did not remove this line, i just splitted on method into two and moved the line here https://github.com/pulp/pulp_rpm/pull/688/files#diff-5185b3f1e5a02c40a5d198fe779c75fbR321